### PR TITLE
Minor cleanup

### PR DESCRIPTION
**Pull request by jrbourbeau**
Just a few small cleanup changes

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -32,9 +32,6 @@ jobs:
           # FIXME: Building Coiled software environments is currently broken on Windows
           - os: windows-latest
             runtime-version: "latest"
-          # TODO: This can be removed once there's a `coiled-runtime=0.0.3` build with `python>=3.7`
-          - python-version: "3.7"
-            runtime-version: "0.0.3"
```
Comment (jrbourbeau) [inline .github/workflows/tests.yml:37]: This exists now

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -14,8 +14,8 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
-  conda:
-    name: Build (and upload) - ${{ matrix.os }}, Python ${{ matrix.python-version }}, Runtime ${{ matrix.runtime-version }}
+  tests:
+    name: ${{ matrix.os }}, Python ${{ matrix.python-version }}, Runtime ${{ matrix.runtime-version }}
```
Comment (jrbourbeau) [inline .github/workflows/tests.yml:18]: This makes the title for PR checks nicer. Currently they look like

```
Conda build / Build (and upload) - ubuntu-latest, Python 3.7, Runtime latest
```

with the changes here they look like

```
Tests / ubuntu-latest, Python 3.7, Runtime latest
```

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -134,7 +131,12 @@ jobs:
 
       - name: Upload conda package
         # Only upload on a single CI build when pushing a tagged commit to `main`
-        if: matrix.os == 'ubuntu-latest' && matrix.python-version == '3.9' && matrix.runtime-version == 'latest' && github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
+        if: |
+          matrix.os == 'ubuntu-latest'
+          && matrix.python-version == '3.9'
+          && matrix.runtime-version == 'latest'
+          && github.event_name == 'push'
+          && startsWith(github.ref, 'refs/tags')
```
Comment (jrbourbeau) [inline .github/workflows/tests.yml:139]: Strict a formatting change (which I think is a little easier to read)

**Review by ncclementi — APPROVED** [review]
These edits are very nice, thanks for the work @jrbourbeau 
I'll merge it as soon as the CI finishes

**Pull request merged**